### ISSUE TITLE
Doesn't work over WAN

### ISSUE THREAD

**Issue by rilespotter** [Schamper/xbox-remote-power#37]
Hi I have been testing and using the Python Script. I could get it to work on LAN but not over a WAN. Any ways to fix this? I was trying to turn on brothers Xbox over WAN. Port Forwarding is done and correct IP address and Live Device ID.

**Comment by tobadia** [comment]
I can confirm this. And also that it worked perfect y over WAN until "a couple months ago".

What bothers me is that I wiresharked the packets exchanged between console and laptop on LAN: laptop sent 4-5 times the same wake-up payload, and Xbox eventually replied on the same port with a readout from its live ID etc. Over WAN, the same packets go in, are properly broadcasted by the router... But console doesn't turn on. This is really puzzling. 

/e: I pushed investigations a little further and forwarded port 5050 to another laptop instead, running wireshark as well. Using laptop 1 (WAN) to send the payload results in laptop 2 (LAN) receiving it unaltered (compared to what is exchanged over LAN) so it doesn't seem to be a broadcast issue by my router.

Any idea @Schamper ? To the best of your knowledge has there been recent changes in the smartglass protocol that could explain this ? Last successful attempt was sometimes in February this year, no changes in network setup since.

**Comment by tobadia** [comment]
Some updates. I can confirm it's an actual change in the SmartGlass protocol that now prevents any "remote" access from anywhere that's not on LAN.
Streaming the console over WAN may no longer be possible.

I am able to power-on the console and detect it from a computer connecter with a VPN (bridged), so that laptop and console are on the same subnet. However streaming won't work for some reason, and it does not seem to be just a matter of redirecting ports.

**Comment by Sunny99** [comment]
So glad I checked here, I've been using this for a while (up until maybe 4 months ago) when it just stopped working without any network alterations. 

I assumed it had to be related to a change of protocols but couldn't confirm it until I saw your posts. Hopefully someone smarter than me can find a solution! I miss being able to game from work and I shudder at the thought of leaving on a dedicated machine to do it!

**Comment by tobadia** [comment]
Don't think there is a "protocol" solution. Most likely the console is now set to ignore any SmartGlass API call coming from a subnet different from its local one. Discussed that a bit on the OpenXbox Discord, apparently the prior imprementation of the API was full of security holes, so I doubt they would roll back. VPN it is, then.

**Comment by Sunny99** [comment]
Ah that's a shame, well I have my workaround - I have a webserver running on a Rpi which calls a modified version of this script (after user validation ofc!) my Rpi is always on so that should sort that issue for me.

Now to see if WAN streaming is still working :/

Edit: So I just tethered my surface pro to my mobile data and remote game streaming is still intact!

**Comment by tobadia** [comment]
I got it working over VPN the other day, but not sure about your setup. Your RPi is on the same LAN as you console, obviously... but where was your Surface ? WAN ?

**Comment by Sunny99** [comment]
yeah my RPi is within the same LAN as my xbox and just used for prompting boot up. My surface when I tested the above edit was tethered to my mobile (cell) phone data (so on a separate network to the console) and streaming was working. However, I'm at the office today (WAN connection), my RPi > Xbox boot up seems to have worked but my xbox reports as being powered down on the xbox app so I need to test it a little more as I think I disabled some forwarded ports the other night which is likely the cause but if you have tested WAN game streaming the other day then that settles it.

**Comment by Sunny99** [comment]
After a little fiddling, I think you are right. I can't get WAN streaming to work either, probably because of the reasons you posted earlier... I had tethered my phone but forgot to turn off my phones wifi so it was sharing its subnet...Doh!

**Comment by tobadia** [comment]
I didn't have much trouble to get it to work once I figured that subset thing. Exact same setting as yours (surface tethered to my mobile hotspot) + VPN to my home router with the usual port forwarding to the Xbox.

Edit: multicast over VPN is a bit of a pain so you can't "discover" the console through the app (at least didn't work during my tests). Not very important though 

The VPN is a bridged OpenVPN, allowing my router to give the client a local IP on the same subnet as its LAN. They xbox app works flawlessly from there.

**Comment by Sunny99** [comment]
Yeah I haven't dabbled with VPNs all that much so I'll check it out, thank you for the information.
